### PR TITLE
logs: refactor logs to connect all actions with the same request_id

### PR DESCRIPTION
**Pull request by grvcoelho**
## Description

Connect all actions with the same requestId on logs. We now accept a `x-request-id` header on `/POST /boletos` that we're going to use to keep track of a given request.

## Your checklist for this pull request

:rotating_light: Please review this items for a good pull request. :four_leaf_clover:

1. I've read the project's [Contributing Guidelines](../CONTRIBUTING.md)
1. My commits are well written and follow [pagarme/git-style-guide](https://github.com/pagarme/git-style-guide)
1. My changes are well covered by **tests** and **logs**
1. I've updated the project docs (if needed)
1. I fell safe about this implementation
1. I feel comfortable with the code I wrote, and I'm not ashamed to show it to my friends

In a good pull request, everything above is true :relaxed:

### CODE REVIEW

File: src/resources/boleto/index.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const create = (event, context, callback) => {
   configureContext(context)
 
   const body = JSON.parse(event.body || JSON.stringify({}))
-  const logger = makeLogger({ operation: 'create' }, { id: defaultCuidValue('req_')() })
```
Comment (wilkmaia) [inline src/resources/boleto/index.ts]: `requestId` não foi definido.

File: src/resources/boleto/index.ts [hunk 2]
```diff
@@ -234,11 +237,9 @@ export const show = (event, context, callback) => {
 export const processBoletosToRegister = (event, context, callback) => {
   configureContext(context)
 
-  const logger = makeLogger(
-    { operation: 'processBoletosToRegister' },{ id: defaultCuidValue('req_')() }
-  )
+  const logger = makeLogger({ operation: 'process_background_queue' },{ id: requestId })
```
Comment (wilkmaia) [inline src/resources/boleto/index.ts]: Acrescenta um espaço após a vírgula.

File: src/providers/bradesco/index.ts [hunk 3]
```diff
@@ -63,17 +63,8 @@ export const buildPayload = boleto =>
     .then(spec => applySpec(spec)(boleto))
 
 export const translateResponseCode = (response) => {
-  const logger = makeLogger({ operation: 'translateResponseCode' })
-
   const responseCode = response.data.status.codigo.toString()
 
-  logger.info({
```
Comment (jeanaugustoos) [inline src/providers/bradesco/index.ts]: Por que tirar o `logger.info`?

Comment (grvcoelho) [reply src/providers/bradesco/index.ts]: Eu movi esse log para a promise chain do método `register` para aproveitar o `requestId` da closure. Assim, a gente não precisa ter a função `translateResponseCode` dentro da função `getProvider` :)

Comment (jeanaugustoos) [reply src/providers/bradesco/index.ts]: Boa! Vi nos commits posteriores :+1:

File: src/providers/bradesco/index.ts [hunk 4]
```diff
@@ -109,6 +106,11 @@ export const getProvider = ({ requestId } = defaultOptions) => {
         })
       })
       .then(axios.request)
+      .tap(response => logger.info({
```
Comment (wilkmaia) [inline src/providers/bradesco/index.ts]: Não é válido acrescentar o `requestId` nesse log?

Comment (jeanaugustoos) [reply src/providers/bradesco/index.ts]: O `requestId` foi adicionado no `logger`, então é exibido neste log.

**Pull request merged**